carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -1378,6 +1379,61 @@ namespace carma_wm
     return curr_light;
   }
 
+   bool CARMAWorldModel::check_if_seen_before_movement_state(boost::posix_time::ptime min_end_time_dynamic,lanelet::CarmaTrafficSignalState received_state_dynamic,uint16_t mov_id, uint8_t mov_signal_group)
+    {
+
+      if(sim_.traffic_signal_states_[mov_id][mov_signal_group].empty())
+      {
+          return 0;
+      }
+
+      for(auto mov_check:sim_.traffic_signal_states_[mov_id][mov_signal_group])
+      {
+
+        auto last_time_difference = mov_check.first - min_end_time_dynamic;  
+        bool is_duplicate = last_time_difference.total_milliseconds() >= -500 && last_time_difference.total_milliseconds() <= 500;
+
+        if(received_state_dynamic == mov_check.second && is_duplicate)
+        {
+          return 1;
+        }
+        else
+        {
+          return 0;
+        }
```

Comment:
this will prematurely return false only checking the first element. return 0 should after when the for loop ended. Also it might be more consistent with rest of the code to return `true`, `false` instead of 0, 1